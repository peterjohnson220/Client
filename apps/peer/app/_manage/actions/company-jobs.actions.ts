import { Action } from '@ngrx/store';
import { GridDataResult } from '@progress/kendo-angular-grid';

import { UpsertExchangeJobMapRequest } from 'libs/models/peer/requests/upsert-exchange-job-map.request.model';
import { CompanyJob } from 'libs/features/peer/job-association/models/company-job.model';
import { ExchangeJob } from 'libs/features/peer/job-association/models/exchange-job.model';

// main grid, misc
export const RESET = '[Peer Manage/Company Jobs] Reset';
export const SET_EXCHANGE_ID = '[Peer Manage/Company Jobs] Set Exchange Id';
export const LOAD_COMPANY_JOBS = '[Peer Manage/Company Jobs] Load Company Jobs';
export const LOAD_COMPANY_JOBS_SUCCESS = '[Peer Manage/Company Jobs] Load Company Jobs Success';
export const LOAD_COMPANY_JOBS_ERROR = '[Peer Manage/Company Jobs] Load Company Jobs Error';
export const LOAD_COMPANY_JOBS_PAGING_ERROR = '[Peer Manage/Company Jobs] Load Company Jobs Paging Error';
export const UPDATE_COMPANY_JOBS_SEARCH_TERM = '[Peer Manage/Company Jobs] Update Company Jobs Search Term';
export const SET_SELECTED_COMPANY_JOB = '[Peer Manage/Company Jobs] Set Selected Company Job';
export const UPDATE_PAGE_ROW_INDEX_TO_SCROLL_TO = '[Peer Manage/Company Jobs] Update Page Row Index To Scroll To';

// exchange job mapped to selected company job
export const LOAD_MAPPED_EXCHANGE_JOBS = '[Peer Manage/Company Jobs] Load Mapped Exchange Jobs';
export const LOAD_MAPPED_EXCHANGE_JOBS_SUCCESS = '[Peer Manage/Company Jobs] Load Mapped Exchange Jobs Success';
export const LOAD_MAPPED_EXCHANGE_JOBS_ERROR = '[Peer Manage/Company Jobs] Load Mapped Exchange Jobs Error';

// jdm PDF Download actions
export const LOAD_JDM_DESCRIPTION_IDS = '[Peer Manage/Company Jobs] Load Jdm Description Ids';
export const LOAD_JDM_DESCRIPTION_IDS_COMPLETE = '[Peer Manage/Company Jobs] Load Jdm Description Ids Complete';
export const DOWNLOAD_JDM_DESCRIPTION = '[Peer Job Manage Jobs] Download Jdm Description';
export const DOWNLOAD_JDM_DESCRIPTION_SUCCESS = '[Peer Manage/Company Jobs] Download Jdm Description Success';
export const DOWNLOAD_JDM_DESCRIPTION_ERROR = '[Peer Manage/Company Jobs] Download Jdm Description Error';

// exchange job search
export const SEARCH_EXCHANGE_JOBS = '[Peer Manage/Company Jobs] Search Exchange Jobs';
export const SEARCH_EXCHANGE_JOBS_SUCCESS = '[Peer Manage/Company Jobs] Search Exchange Jobs Success';
export const SEARCH_EXCHANGE_JOBS_ERROR = '[Peer Manage/Company Jobs] Search Exchange Jobs Error';
export const UPDATE_EXCHANGE_JOBS_TITLE_SEARCH_TERM = '[Peer Manage/Company Jobs] Update Exchange Jobs Title Search Term';
export const UPDATE_EXCHANGE_JOBS_DESCRIPTION_SEARCH_TERM = '[Peer Manage/Company Jobs] Update Exchange Jobs Description Search Term';

// exchange job search, create new association
export const CREATE_ASSOCIATION = '[Peer Manage/Company Jobs] Create Association';
export const CREATE_ASSOCIATION_SUCCESS = '[Peer Manage/Company Jobs] Create Association Success';
export const CREATE_ASSOCIATION_ERROR = '[Peer Manage/Company Jobs] Create Association Error';

// exchange job search, approve/reject/unmatch
export const APPROVE_PENDING_MATCH = '[Peer Manage/Company Jobs] Approve Pending Match';
export const APPROVE_PENDING_MATCH_SUCCESS = '[Peer Manage/Company Jobs] Approve Pending Match Success';
export const APPROVE_PENDING_MATCH_ERROR = '[Peer Manage/Company Jobs] Approve Pending Match Error';
export const UNMATCH = '[Peer Manage/Company Jobs] Unmatch';
export const UNMATCH_SUCCESS = '[Peer Manage/Company Jobs] Unmatch Success';
export const UNMATCH_ERROR = '[Peer Manage/Company Jobs] Unmatch Error';
export const CONFIRM_UNMATCH = '[Peer Manage/Company Jobs] Confirm Unmatch';
export const CANCEL_UNMATCH = '[Peer Manage/Company Jobs] Cancel Unmatch';

export class Reset implements Action {
  readonly type = RESET;
}

export class SetExchangeId implements Action {
  readonly type = SET_EXCHANGE_ID;
  constructor(public payload: number) {}
}

export class LoadCompanyJobs implements Action {
  readonly type = LOAD_COMPANY_JOBS;
}

export class LoadCompanyJobsSuccess implements Action {
  readonly type = LOAD_COMPANY_JOBS_SUCCESS;
  constructor(public payload: GridDataResult) {}
}

export class LoadCompanyJobsError implements Action {
  readonly type = LOAD_COMPANY_JOBS_ERROR;
}

export class LoadCompanyJobsPagingError implements Action {
  readonly type = LOAD_COMPANY_JOBS_PAGING_ERROR;
  constructor(public payload: string) {}
}

export class UpdateCompanyJobsSearchTerm implements Action {
  readonly type = UPDATE_COMPANY_JOBS_SEARCH_TERM;
  constructor(public payload: string) {}
}

export class SetSelectedCompanyJob implements Action {
  readonly type = SET_SELECTED_COMPANY_JOB;
  constructor(public payload: CompanyJob) {}
}

export class UpdatePageRowIndexToScrollTo implements Action {
  readonly type = UPDATE_PAGE_ROW_INDEX_TO_SCROLL_TO;
  constructor(public payload: number) {}
}

export class LoadMappedExchangeJobs implements Action {
  readonly type = LOAD_MAPPED_EXCHANGE_JOBS;
  constructor(public payload: number) {}
}

export class LoadMappedExchangeJobsSuccess implements Action {
  readonly type = LOAD_MAPPED_EXCHANGE_JOBS_SUCCESS;
  constructor(public payload: ExchangeJob[]) {}
}

export class LoadMappedExchangeJobsError implements Action {
  readonly type = LOAD_MAPPED_EXCHANGE_JOBS_ERROR;
}

// jdm PDF download
export class LoadJdmDescriptionIds implements Action {
  readonly type = LOAD_JDM_DESCRIPTION_IDS;
}

export class LoadJdmDescriptionIdsComplete implements Action {
  readonly type = LOAD_JDM_DESCRIPTION_IDS_COMPLETE;
  constructor(public payload: number[]) {}
}

export class DownloadJdmDescription implements Action {
  readonly type = DOWNLOAD_JDM_DESCRIPTION;
}

export class DownloadJdmDescriptionSuccess implements Action {
  readonly type = DOWNLOAD_JDM_DESCRIPTION_SUCCESS;
}

export class DownloadJdmDescriptionError implements Action {
  readonly type = DOWNLOAD_JDM_DESCRIPTION_ERROR;
}

// exchange job search
export class SearchExchangeJobs implements Action {
  readonly type = SEARCH_EXCHANGE_JOBS;
}

export class SearchExchangeJobsSuccess implements Action {
  readonly type = SEARCH_EXCHANGE_JOBS_SUCCESS;
  constructor(public payload: ExchangeJob[]) {}
}

export class SearchExchangeJobsError implements Action {
  readonly type = SEARCH_EXCHANGE_JOBS_ERROR;
}

export class UpdateExchangeJobsTitleSearchTerm implements Action {
  readonly type = UPDATE_EXCHANGE_JOBS_TITLE_SEARCH_TERM;
  constructor(public payload: string) {}
}

export class UpdateExchangeJobsDescriptionSearchTerm implements Action {
  readonly type = UPDATE_EXCHANGE_JOBS_DESCRIPTION_SEARCH_TERM;
  constructor(public payload: string) {}
}

export class CreateAssociation implements Action {
  readonly type = CREATE_ASSOCIATION;
  constructor(public payload: UpsertExchangeJobMapRequest) {}
}

export class CreateAssociationSuccess implements Action {
  readonly type = CREATE_ASSOCIATION_SUCCESS;
}

export class CreateAssociationError implements Action {
  readonly type = CREATE_ASSOCIATION_ERROR;
}

export class ApprovePendingMatch implements Action {
  readonly type = APPROVE_PENDING_MATCH;
  constructor(public payload: UpsertExchangeJobMapRequest) {}
}

export class ApprovePendingMatchSuccess implements Action {
  readonly type = APPROVE_PENDING_MATCH_SUCCESS;
}

export class ApprovePendingMatchError implements Action {
  readonly type = APPROVE_PENDING_MATCH_ERROR;
}

export class Unmatch implements Action {
  readonly type = UNMATCH;
  constructor(public payload: UpsertExchangeJobMapRequest) {}
}

export class UnmatchSuccess implements Action {
  readonly type = UNMATCH_SUCCESS;
}

export class UnmatchError implements Action {
  readonly type = UNMATCH_ERROR;
}

export class ConfirmUnmatch implements Action {
  readonly type = CONFIRM_UNMATCH;
}

export class CancelUnmatch implements Action {
  readonly type = CANCEL_UNMATCH;
}

export type Actions =
  // main grid, misc
  | Reset
  | SetExchangeId
  | LoadCompanyJobs
  | LoadCompanyJobsSuccess
  | LoadCompanyJobsError
  | LoadCompanyJobsPagingError
  | UpdateCompanyJobsSearchTerm
  | SetSelectedCompanyJob
  | UpdatePageRowIndexToScrollTo
  // mapped exchange job in detail panel
  | LoadMappedExchangeJobs
  | LoadMappedExchangeJobsSuccess
  | LoadMappedExchangeJobsError
  // jdm PDF download
  | LoadJdmDescriptionIds
  | LoadJdmDescriptionIdsComplete
  | DownloadJdmDescription
  | DownloadJdmDescriptionSuccess
  | DownloadJdmDescriptionError
  // exchange job search
  | SearchExchangeJobs
  | SearchExchangeJobsSuccess
  | SearchExchangeJobsError
  | UpdateExchangeJobsTitleSearchTerm
  | UpdateExchangeJobsDescriptionSearchTerm
  // create match
  | CreateAssociation
  | CreateAssociationSuccess
  | CreateAssociationError
  // approve/deny/unmatch
  | ApprovePendingMatch
  | ApprovePendingMatchSuccess
  | ApprovePendingMatchError
  | Unmatch
  | UnmatchSuccess
  | UnmatchError
  | ConfirmUnmatch
  | CancelUnmatch;
