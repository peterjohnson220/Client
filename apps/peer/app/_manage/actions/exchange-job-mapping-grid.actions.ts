import { Action } from '@ngrx/store';
import { GridDataResult } from '@progress/kendo-angular-grid';

import { ExchangeJobMapping } from 'libs/models/peer';

export const LOAD_EXCHANGE_JOB_MAPPINGS  = '[Peer Main/Exchange Job Mapping] Load Exchange Job Mappings';
export const LOAD_EXCHANGE_JOB_MAPPINGS_SUCCESS  = '[Peer Main/Exchange Job Mapping] Load Exchange Job Mappings Success';
export const LOAD_EXCHANGE_JOB_MAPPINGS_ERROR = '[Peer Main/Exchange Job Mapping] Load Exchange Job Mappings Error';
export const SELECT_EXCHANGE_JOB_MAPPING = '[Peer Main/Exchange Job Mapping] Select Exchange Job Mapping';
export const RESELECT_EXCHANGE_JOB_MAPPING = '[Peer Main/Exchange Job Mapping] ReSelect Exchange Job Mapping';
export const UPDATE_PAGE_ROW_INDEX_TO_SCROLL_TO = '[Peer Main/Exchange Job Mapping] Update Page Row Index To Scroll To';
export const LOAD_EXCHANGE_JOB_MAPPINGS_AFTER_MAP  = '[Peer Main/Exchange Job Mapping] Load Exchange Job Mappings After Map';

export class LoadExchangeJobMappings implements Action {
  readonly type = LOAD_EXCHANGE_JOB_MAPPINGS;

  constructor(public payload: any) {}
}

export class LoadExchangeJobMappingsSuccess implements Action {
  readonly type = LOAD_EXCHANGE_JOB_MAPPINGS_SUCCESS;

  constructor(public payload: GridDataResult) {}
}

export class LoadExchangeJobMappingsError implements Action {
  readonly type = LOAD_EXCHANGE_JOB_MAPPINGS_ERROR;
}

export class SelectExchangeJobMapping implements Action {
  readonly type = SELECT_EXCHANGE_JOB_MAPPING;

  constructor(public payload: ExchangeJobMapping) {}
}

export class UpdatePageRowIndexToScrollTo implements Action {
  readonly type = UPDATE_PAGE_ROW_INDEX_TO_SCROLL_TO;

  constructor(public payload: number) {}
}

export class ReSelectExchangeJobMapping implements Action {
  readonly type = RESELECT_EXCHANGE_JOB_MAPPING;
}

export class LoadExchangeJobMappingsAfterMap implements Action {
  readonly type = LOAD_EXCHANGE_JOB_MAPPINGS_AFTER_MAP;
}

export type Actions
  = LoadExchangeJobMappings
  | LoadExchangeJobMappingsSuccess
  | LoadExchangeJobMappingsError
  | SelectExchangeJobMapping
  | UpdatePageRowIndexToScrollTo
  | ReSelectExchangeJobMapping
  | LoadExchangeJobMappingsAfterMap;
