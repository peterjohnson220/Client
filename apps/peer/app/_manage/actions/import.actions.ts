import { Action } from '@ngrx/store';

export const SWITCH_ASSOCIATION_IMPORT_MODAL_OPEN = '[Peer Main/Exchange Request] Switch Association Import Modal';
export const GET_ASSOCIATION_IMPORT_MODAL = '[Peer Main/Exchange Request] Get Association Import Modal';
export const SWITCH_ASSOCIATION_IMPORT_STATUS = '[Peer Main/Exchange Request] Switch Associations Import Status';
export const GET_ASSOCIATION_IMPORT_STATUS = '[Peer Main/Exchange Request] Get Associations Import Status';

export class SwitchAssociationImportModalOpenAction implements Action {
    readonly type = SWITCH_ASSOCIATION_IMPORT_MODAL_OPEN;
    constructor(public payload: boolean) { }
}

export class SwitchAssociationImportStatus implements Action {
    readonly type = SWITCH_ASSOCIATION_IMPORT_STATUS;
    constructor(public payload: ImportStatusEnum) { }
}

export class GetAssociationImportModalAction implements Action {
    readonly type = GET_ASSOCIATION_IMPORT_MODAL;
}

export class GetAssociationImportStatus implements Action {
    readonly type = GET_ASSOCIATION_IMPORT_STATUS;
}

export type Actions =
    SwitchAssociationImportModalOpenAction |
    GetAssociationImportModalAction |
    SwitchAssociationImportStatus |
    GetAssociationImportStatus;

export enum ImportStatusEnum {
    Failed = -1,
    Idle = 0,
    Success = 1,
    InProcess = 2
}
