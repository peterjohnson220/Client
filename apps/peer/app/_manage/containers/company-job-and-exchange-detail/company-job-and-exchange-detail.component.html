<section class="card mb-2">
  <div class="card-header">
    <h6 class="text-uppercase mb-0">Company Job and Exchange</h6>
  </div>
  <a href="javascript:void(0)" (click)="handleCloseClick()" class="close-panel mr-3 mt-1" title="Close">
    <fa-icon icon="times"></fa-icon>
  </a>
  <div class="d-flex">
    <div class="w-50 p-3">
      <pf-company-job-detail 
        [selectedCompanyJob]="selectedCompanyJob$ | async"
        [jdmDescriptionIds]="jdmDescriptionIds$ | async"
        [jdmDescriptionLoading]="jdmDescriptionLoading$ | async"
        [jdmDescriptionLoadingError]="jdmDescriptionLoadingError$ | async" 
        (viewJdmDescriptionClick)="handleViewJdmDescriptionClick()">
      </pf-company-job-detail>
    </div>
    <span class="bg-image">
      <fa-icon icon="exchange"></fa-icon>
    </span>
    <div class="w-50 p-3">
      <pf-exchange-job-detail
        *ngIf="(selectedCompanyJob$ | async) && (selectedCompanyJob$ | async).IsAssociated; else exchangeJobSearch"
        [exchangeJob]="mappedExchangeJob$ | async"
        [exchangeJobLoading]="mappedExchangeJobsLoading$ | async"
        [exchangeJobLoadingSuccess]="mappedExchangeJobsLoadingSuccess$ | async"
        [exchangeJobLoadingError]="mappedExchangeJobsLoadingError$ | async">
      </pf-exchange-job-detail>
      <ng-template #exchangeJobSearch>
        <pf-exchange-job-search ></pf-exchange-job-search>
      </ng-template>
    </div>
  </div>
</section>