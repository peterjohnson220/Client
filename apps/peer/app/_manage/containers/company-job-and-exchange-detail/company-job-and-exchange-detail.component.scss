@import 'colors';
a.close-panel {
  position: absolute;
  right: 0;
  opacity: .85;
  transition: opacity .3s ease-in-out;
  color: #333;
  &:hover{
    opacity: 1;
  }
  top: 0;
}

.job-information {
  height: calc(100vh - 385px);
}
