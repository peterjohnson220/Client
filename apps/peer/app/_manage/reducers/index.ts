import { createSelector, createFeatureSelector } from '@ngrx/store';

// Import root app reducer
import * as fromRoot from 'libs/state/state';

// Import feature reducers
import * as fromGridReducer from 'libs/core/reducers/grid.reducer';
import { IFeatureGridState } from 'libs/core/reducers/grid.reducer';

import * as fromExchangeJobMappingGridReducer from './exchange-job-mapping-grid.reducer';
import * as fromExchangeJobMappingInfoReducer from './exchange-job-mapping-info.reducer';
import * as fromPfJobsExchangeRequestReducer from './pf-jobs.reducer';
import * as fromSharedPeerReducer from '../../shared/reducers';
import * as fromExchangeRequestReducer from '../../shared/reducers/exchange-request.reducer';
import * as fromJobFamiliesReducer from './job-families.reducer';

// Feature area state
export interface PeerManagementState {
  exchangeJobMapping: IFeatureGridState<fromExchangeJobMappingGridReducer.State>;
  exchangeJobMappingInfo: fromExchangeJobMappingInfoReducer.State;
  pfJobsExchangeRequest: fromPfJobsExchangeRequestReducer.State;
  jobFamilies: fromJobFamiliesReducer.State;
}

// Extend root state with feature area state
export interface State extends fromRoot.State {
  peer_manage: PeerManagementState;
}

// Feature area reducers
export const reducers = {
  exchangeJobMapping: fromExchangeJobMappingGridReducer.reducer,
  exchangeJobMappingInfo: fromExchangeJobMappingInfoReducer.reducer,
  pfJobsExchangeRequest: fromPfJobsExchangeRequestReducer.reducer,
  jobFamilies: fromJobFamiliesReducer.reducer
};

// Select Feature Area
export const selectFeatureAreaState = createFeatureSelector<PeerManagementState>('peer_manage');

// Feature Selectors
export const selectExchangeJobMappingState = createSelector(
  selectFeatureAreaState,
  (state: PeerManagementState) => state.exchangeJobMapping
);

export const selectExchangeJobMappingInfoState = createSelector(
  selectFeatureAreaState,
  (state: PeerManagementState) => state.exchangeJobMappingInfo
);

export const selectPfJobsExchangeRequestState = createSelector(
  selectFeatureAreaState,
  (state: PeerManagementState) => state.pfJobsExchangeRequest
);

export const selectJobFamiliesState = createSelector(
  selectFeatureAreaState,
  (state: PeerManagementState) => state.jobFamilies
);

// Exchange Job Mapping Selectors
export const selectExchangeJobMappingsFeatureState = createSelector(
  selectExchangeJobMappingState,
  (state: IFeatureGridState<fromExchangeJobMappingGridReducer.State>) => state.feature
);

export const selectExchangeJobMappingsGridState = createSelector(
  selectExchangeJobMappingState,
  (state: IFeatureGridState<fromExchangeJobMappingGridReducer.State>) => state.grid
);

export const {
  selectAll: getExchangeJobMappings
} = fromExchangeJobMappingGridReducer.adapter.getSelectors(selectExchangeJobMappingsFeatureState);

export const getExchangeJobMappingsLoading = createSelector(
  selectExchangeJobMappingsFeatureState,
  fromExchangeJobMappingGridReducer.getLoading
);

export const getExchangeJobMappingsLoadingError = createSelector(
  selectExchangeJobMappingsFeatureState,
  fromExchangeJobMappingGridReducer.getLoadingError
);

export const getExchangeJobMappingsTotal = createSelector(
  selectExchangeJobMappingsFeatureState,
  fromExchangeJobMappingGridReducer.getTotal
);

export const getExchangeJobMappingsGridState = createSelector(
  selectExchangeJobMappingsGridState,
  fromGridReducer.getGridState
);

export const getSelectedExchangeJobMapping = createSelector(
  selectExchangeJobMappingsFeatureState,
  fromExchangeJobMappingGridReducer.getSelectedMapping
);

export const getExchangeJobMappingPageRowIndexToScrollTo = createSelector(
  selectExchangeJobMappingsFeatureState,
  fromExchangeJobMappingGridReducer.getPageRowIndexToScrollTo
);

export const getExchangeJobMappingsGridData = createSelector(
  getExchangeJobMappings,
  getExchangeJobMappingsTotal,
  (data, total) => {
    return { data: data, total: total };
  }
);

export const getLoadExchangeJobMappingGridRequest = createSelector(
  fromSharedPeerReducer.getExchange,
  getExchangeJobMappingsGridState,
  (exchange, gridState) => {
    return { exchangeId: exchange ? exchange.ExchangeId : 0, listState: gridState };
  }
);

// Exchange Job Mapping Info Selectors
export const getCompanyJobsToMapToLoading = createSelector(
  selectExchangeJobMappingInfoState,
  fromExchangeJobMappingInfoReducer.getLoading
);

export const getCompanyJobsToMapToLoadingError = createSelector(
  selectExchangeJobMappingInfoState,
  fromExchangeJobMappingInfoReducer.getLoadingError
);

export const getExchangeJobsInfoApplyingMapping = createSelector(
  selectExchangeJobMappingInfoState,
  fromExchangeJobMappingInfoReducer.getApplyingMapping
);

export const getExchangeJobsInfoApplyingMappingError = createSelector(
  selectExchangeJobMappingInfoState,
  fromExchangeJobMappingInfoReducer.getApplyingMappingError
);

export const getExchangeJobsInfoSelectedMappingCompanyJobId = createSelector(
  selectExchangeJobMappingInfoState,
  fromExchangeJobMappingInfoReducer.getSelectedMappingCompanyJobId
);

export const getExchangeJobsInfoEditingMapping = createSelector(
  selectExchangeJobMappingInfoState,
  fromExchangeJobMappingInfoReducer.getEditingMapping
);

export const {
  selectAll: getCompanyJobsToMapTo
} = fromExchangeJobMappingInfoReducer.adapter.getSelectors(selectExchangeJobMappingInfoState);

// Exchange Request - pfJobs
export const {
  selectAll: getPfJobsExchangeRequestCandidates
} = fromPfJobsExchangeRequestReducer.adapter.getSelectors(selectPfJobsExchangeRequestState);
export const getPfJobsExchangeRequestCandidatesLoading = createSelector(
  selectPfJobsExchangeRequestState,
  fromExchangeRequestReducer.getLoading
);
export const getPfJobsExchangeRequestCandidatesLoadingError = createSelector(
  selectPfJobsExchangeRequestState,
  fromExchangeRequestReducer.getLoadingError
);
export const getPfJobsExchangeRequestModalOpen = createSelector(
  selectPfJobsExchangeRequestState,
  fromExchangeRequestReducer.getModalOpen
);
export const getPfJobsExchangeRequestRequesting = createSelector(
  selectPfJobsExchangeRequestState,
  fromExchangeRequestReducer.getRequesting
);
export const getPfJobsExchangeRequestContext = createSelector(
  selectPfJobsExchangeRequestState,
  fromSharedPeerReducer.getExchange,
  (exchangeRequestState, exchange) => {
    return {
      Query: exchangeRequestState.searchTerm,
      ExchangeId: exchange ? exchange.ExchangeId : 0,
      JobDescriptionQuery: exchangeRequestState.filterOptions ? exchangeRequestState.filterOptions.JobDescriptionQuery : ''
    };
  }
);

// Job Families
export const {
  selectAll: getJobFamilies
} = fromJobFamiliesReducer.adapter.getSelectors(selectJobFamiliesState);
export const getJobFamiliesLoading = createSelector(
  selectJobFamiliesState,
  fromJobFamiliesReducer.getLoading
);
export const getJobFamiliesLoadingError = createSelector(
  selectJobFamiliesState,
  fromJobFamiliesReducer.getLoadingError
);
