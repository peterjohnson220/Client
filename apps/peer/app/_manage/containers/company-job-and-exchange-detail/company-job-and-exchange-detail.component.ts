import { Component, OnInit } from '@angular/core';

import { select, Store } from '@ngrx/store';
import { Observable } from 'rxjs';

import { CompanyJob } from 'libs/models/company';
import { ExchangeJob } from 'libs/features/peer/job-association/models/exchange-job.model';

import * as companyJobsActions from '../../actions/company-jobs.actions';
import * as companyJobsReducer from '../../reducers';

@Component({
  selector: 'pf-company-job-and-exchange-detail',
  templateUrl: './company-job-and-exchange-detail.component.html',
  styleUrls: ['./company-job-and-exchange-detail.component.scss']
})
export class CompanyJobAndExchangeDetailComponent implements OnInit {
  selectedCompanyJob$: Observable<CompanyJob>;

  mappedExchangeJobsLoading$: Observable<boolean>;
  mappedExchangeJobsLoadingSuccess$: Observable<boolean>;
  mappedExchangeJobsLoadingError$: Observable<boolean>;
  mappedExchangeJob$: Observable<ExchangeJob>;

  jdmDescriptionIds$: Observable<number[]>;
  jdmDescriptionLoading$: Observable<boolean>;
  jdmDescriptionLoadingError$: Observable<boolean>;

  constructor(private store: Store<companyJobsReducer.State>) { }

  ngOnInit() {
    this.selectedCompanyJob$ = this.store.pipe(select(companyJobsReducer.getCompanyJobsSelectedCompanyJob));

    this.mappedExchangeJobsLoading$ = this.store.pipe(select(companyJobsReducer.getCompanyJobsMappedExchangeJobsLoading));
    this.mappedExchangeJobsLoadingSuccess$ = this.store.pipe(select(companyJobsReducer.getCompanyJobsMappedExchangeJobsLoadingSuccess));
    this.mappedExchangeJobsLoadingError$ = this.store.pipe(select(companyJobsReducer.getCompanyJobsMappedExchangeJobsLoadingError));
    this.mappedExchangeJob$ = this.store.pipe(select(companyJobsReducer.getCompanyJobsMappedExchangeJob));

    this.jdmDescriptionIds$ = this.store.pipe(select(companyJobsReducer.getCompanyJobsJdmDescriptionIds));
    this.jdmDescriptionLoading$ = this.store.pipe(select(companyJobsReducer.getCompanyJobsDownloadingJdmDescription));
    this.jdmDescriptionLoadingError$ = this.store.pipe(select(companyJobsReducer.getCompanyJobsDownloadingJdmDescriptionError));
  }

  handleCloseClick() {
    this.store.dispatch(new companyJobsActions.SetSelectedCompanyJob(null));
  }

  handleViewJdmDescriptionClick() {
    this.store.dispatch(new companyJobsActions.DownloadJdmDescription());
  }
}
