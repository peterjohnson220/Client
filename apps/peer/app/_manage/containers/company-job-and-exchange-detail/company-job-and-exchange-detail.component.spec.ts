import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { NO_ERRORS_SCHEMA } from '@angular/core';

import { combineReducers, Store, StoreModule } from '@ngrx/store';

import * as fromRootState from 'libs/state/state';
import * as companyJobsActions from '../../actions/company-jobs.actions';
import * as manageReducers from '../../reducers/';

import { CompanyJobAndExchangeDetailComponent } from './company-job-and-exchange-detail.component';

describe('CompanyJobAndExchangeDetailComponent', () => {
  let component: CompanyJobAndExchangeDetailComponent;
  let fixture: ComponentFixture<CompanyJobAndExchangeDetailComponent>;
  let store: Store<manageReducers.State>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [
        StoreModule.forRoot({
          ...fromRootState.reducers,
          peer_manage: combineReducers(manageReducers.reducers)
        }),
      ],
      declarations: [ CompanyJobAndExchangeDetailComponent ],
      schemas: [NO_ERRORS_SCHEMA]
    })
    .compileComponents();

    store = TestBed.get(Store);
    spyOn(store, 'dispatch');
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(CompanyJobAndExchangeDetailComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should render a close link plus the company job and exchange job detail panels', () => {
    expect(fixture).toMatchSnapshot();
  });

  it('should dispatch the expected action when the close link is clicked', () => {
    const closePanel = fixture.debugElement.nativeElement.querySelector('a.close-panel');
    closePanel.click();

    const selectCompanyJobAction = new companyJobsActions.SetSelectedCompanyJob(null);

    expect(store.dispatch).toHaveBeenCalledWith(selectCompanyJobAction);
  });
});
