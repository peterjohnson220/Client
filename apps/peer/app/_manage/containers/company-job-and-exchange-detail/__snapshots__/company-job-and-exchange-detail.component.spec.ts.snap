// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompanyJobAndExchangeDetailComponent should render the exchange job detail panel when the selected company job is associated 1`] = `
<pf-company-job-and-exchange-detail
  allSubscriptions={[Function Subscription]}
  exchangeId="null"
  jdmDescriptionIds$={[Function Store]}
  jdmDescriptionLoading$={[Function Store]}
  jdmDescriptionLoadingError$={[Function Store]}
  mappedExchangeJob$={[Function Store]}
  mappedExchangeJobsLoading$={[Function Store]}
  mappedExchangeJobsLoadingError$={[Function Store]}
  mappedExchangeJobsLoadingSuccess$={[Function Store]}
  savingAssociation$={[Function Store]}
  savingAssociationError$={[Function Store]}
  selectedCompanyJob="null"
  selectedCompanyJob$={[Function Observable]}
  store={[Function Store]}
>
  <section
    class="card mb-2"
  >
    <div
      class="card-header"
    >
      <h6
        class="mb-0 font-weight-bold"
      >
        Job Information
      </h6>
    </div>
    <a
      class="close-panel mr-2 mt-1"
      href="javascript:void(0)"
      title="Close"
    >
      <fa-icon
        icon="times"
      />
    </a>
    <div
      class="d-flex"
    >
      <div
        class="w-50 p-3"
      >
        <pf-company-job-detail />
      </div>
      <div
        class="vertical-separator mt-3 mb-2"
      />
      <div
        class="w-50 p-3"
      >
        
        <pf-exchange-job-detail />
        
      </div>
    </div>
  </section>
</pf-company-job-and-exchange-detail>
`;

exports[`CompanyJobAndExchangeDetailComponent should render the exchange job search panel when the selected company job is not associated 1`] = `
<pf-company-job-and-exchange-detail
  allSubscriptions={[Function Subscription]}
  exchangeId="null"
  jdmDescriptionIds$={[Function Store]}
  jdmDescriptionLoading$={[Function Store]}
  jdmDescriptionLoadingError$={[Function Store]}
  mappedExchangeJob$={[Function Store]}
  mappedExchangeJobsLoading$={[Function Store]}
  mappedExchangeJobsLoadingError$={[Function Store]}
  mappedExchangeJobsLoadingSuccess$={[Function Store]}
  savingAssociation$={[Function Store]}
  savingAssociationError$={[Function Store]}
  selectedCompanyJob="null"
  selectedCompanyJob$={[Function Observable]}
  store={[Function Store]}
>
  <section
    class="card mb-2"
  >
    <div
      class="card-header"
    >
      <h6
        class="mb-0 font-weight-bold"
      >
        Job Information
      </h6>
    </div>
    <a
      class="close-panel mr-2 mt-1"
      href="javascript:void(0)"
      title="Close"
    >
      <fa-icon
        icon="times"
      />
    </a>
    <div
      class="d-flex"
    >
      <div
        class="w-50 p-3"
      >
        <pf-company-job-detail />
      </div>
      <div
        class="vertical-separator mt-3 mb-2"
      />
      <div
        class="w-50 p-3"
      >
        
        <pf-exchange-job-search />
        
      </div>
    </div>
  </section>
</pf-company-job-and-exchange-detail>
`;
